core/storage/pager.rs

```diff
@@ -624,7 +721,10 @@ impl Pager {
             // read frame or page
             match page_cache.insert(page_key, page.clone()) {
                 Ok(_) => {}
-                Err(CacheError::Full) => return Err(LimboError::CacheFull),
+                Err(CacheError::Full) => {
+                    drop(page_cache);
+                    self.stress(page.clone())?;
```

Comment:
This probably means `read_page` should be a state machine because stress performs I/O, so next time we call `read_page` we need to check if we are still moving page to wal in `stress`.